wdl/tasks.wdl

```diff
@@ -790,4 +772,56 @@ task PrintReads {
         vcf: {localization_optional: true}
         vcf_idx: {localization_optional: true}
     }
+}
+
+# Ensure ordering of bam contigs are in the same order as ref_dict
+task ReorderSam {
+    input {
+        File? ref_fasta
+        File? ref_fasta_index
+        File ref_dict
+
+        String prefix
+        File bam
+        File bai
+
+        Runtime runtime_params
+    }
+    String reordered_bam = prefix + ".reordered.bam"
+    String reordered_bai = prefix + ".reordered.bai"
+    Int diskGB = runtime_params.disk + ceil(size(bam, "GB") * 2) + ceil(size(ref_dict, "GB"))
+
+    command <<<
+        set -e
+        export GATK_LOCAL_JAR=~{select_first([runtime_params.jar_override, "/root/gatk.jar"])}
+        gatk --java-options "-Xmx~{runtime_params.command_mem}m" \
+            ReorderSam \
+            -I '~{bam}' \
+            -O '~{reordered_bam}' \
+            -SD '~{ref_dict}' \
+            --CREATE_INDEX true
+    >>>
+
+    output {
+        File reordered_bam = reordered_bam
```

Comment:
I don't think this works since in WDL you can only use each variable name once in a context. Overwriting variables is not possible.